base/test.jl

```diff
@@ -22,6 +22,30 @@ export GenericString
 
 #-----------------------------------------------------------------------
 
+# Backtrace utility functions
+function ip_matches_func_and_name(ip, func::Symbol, dir::String, file::String)
+    for fr in StackTraces.lookup(ip)
+        if fr === StackTraces.UNKNOWN || fr.from_c
+            return false
+        end
+        path = string(fr.file)
+        fr.func == func && dirname(path) == dir && basename(path) == file && return true
+    end
+    return false
+end
+
+function scrub_backtrace(bt)
+    do_test_ind = findfirst(addr->ip_matches_func_and_name(addr, :do_test, ".", "test.jl"), bt)
```

Comment:
Will this get confused if the user has a function `do_test` in a file `test.jl`?

(We could always add a random string like `_409b71657771` to the end of our `do_test` function to make this less likely?)